packages/core/src/components/menu/_menu.scss

```diff
@@ -69,6 +71,14 @@ Styleguide menu
       margin-right: $menu-item-padding-large;
     }
   }
+
+  .#{$ns}-small & {
+    @include menu-item-small();
+  }
+}
+
+.#{$ns}-menu-no-padding {
+  padding: 0;
 }
```

Comment:
You don't need to create a new modifier class to make dividers take up full width. You can use negative margins to achieve the same result:

![image](https://github.com/palantir/blueprint/assets/723999/5f9ce0f0-86a0-42d5-84e3-5c06107e25ad)

Also I would prefer not to support _both_ the current MenuDivider visual design _and_ a new variant where the divider takes up full width. I think we should be opinionated and just pick one. If we think full width dividers are the direction we want the visual design to go towards, then let's just change the styles to reflect that and make all MenuDividers full width.